src/components/PromotionList.tsx

```diff
@@ -36,12 +36,17 @@ export default function PromotionList({
    */
   useEffect(() => {
     dispatch({ type: DispatchAction.DATA_LOADING });
+
     filterPromotions(promotionListState.filter)
       .then((filteredPromotions: Promotion[]) =>
         sortPromotions(filteredPromotions, promotionListState.sort)
       )
       .then((sortedPromotions: Promotion[]) => {
         dispatch({ type: DispatchAction.DATA_SUCCESS });
+
+        // If search bar contains keyword, reset it. This is only minimal behaviour, needs work with PP-68
+        dispatch({ type: DispatchAction.RESET_SEARCH_QUERY });
+
```

Comment:
I added this temporarily to handle the case where you do 
1. search bar "fry"
2. choose filter/dropdown
3. search bar "fry" again (same searchQuery)
This ensures that for 3, it gets properly called because before, since the searchQuery didn't change, it actually wouldn't have gotten the promotions.

Reply:
Ohh I see, that makes sense good catch!!